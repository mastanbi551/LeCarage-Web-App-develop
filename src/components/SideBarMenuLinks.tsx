import React from 'react';
import { FiHome, FiUser } from 'react-icons/fi';
import { RiCalendarTodoFill } from 'react-icons/ri';
import {SlCalender} from 'react-icons/sl';

export const SideBarMenuLinks = [
    {
        'path': '/',
        'name': 'Dashboard',
        'icon': <FiHome />,
    },
    {
        'path': '/usertypes',
        'name': 'User Management',
        'icon': <FiUser />,
    },
    {
        'path': '/usertypes',
        'name': 'User Types',
    },
    {
        'path': '/users',
        'name': 'Users',
      },
      {
        'path': '/customers',
        'name': 'Customers',
      },
      {
        'path': '/cars',
        'name': 'Cars',
      },
      {
        'path': '/BookingsList',
        'name': 'Bookings',
        'icon': <RiCalendarTodoFill />,
      },
      {
        'path': '/BookingsList',
        'name': 'All Bookings',
        // icon: <FaMoneyBill />,
      },
      {
        'path': '/Calender',
        'name': 'Calendar',
        'icon': <SlCalender />,
      },
      {
        'path': '/Calender',
        'name': 'Events'
      },
      {
        'path': 'Services',
        'name': 'Services'
      }
];

export const DefaultMenu = [
    {
        'path': '/',
        'name': 'Dashboard',
        'icon': <FiHome />,
    },
];