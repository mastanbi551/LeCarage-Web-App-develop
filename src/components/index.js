import { SideBar } from './sidebar/SideBar';


export { CustomButton } from './customButton/CustomButton';
export { BackgroundImageContainer } from './backgroundImageContainer/BackgroundImageContainer';
export { LeCarageLogo } from './leCarageLogo/LeCarageLogo';
export { PageHeader } from './pageHeader/PageHeader';
export { InputTextField } from './inputTextField/InputTextField';
export { Header } from './header/Header';
export { CustomDropDown } from './customDropDown/CustomDropDown';
export { CheckBoxList } from './checkBoxList/CheckBoxList';
export { Loader } from './loader/Loader';

export { PageContainer } from './pageContainer/PageContainer';
export { DetailsContainer } from './pageContainer/DetailsContainer';
export { CardContainer } from './pageContainer/CardContainer';


export { DynamicDataTable } from './dynamicDataTable/DynamicDataTable';
export { DynamicDataTableHeader } from './dynamicDataTable/DynamicDataTableHeader';
export { DynamicDataTableItems } from './dynamicDataTable/DynamicDataTableItems';
export { CustomRadioButtonList } from './customRadioButtonsList/CustomRadioButtonsList';
export { CustomFilter } from './customFilter/CustomFilter';

export { GooglePlaces } from '../components/googlePlaces/GooglePlaces';
export { CustomCheckBox } from '../components/customCheckBox/CustomCheckBox';
export { SearchableDropdown } from './searchableDropDown/SearchableDropdown';

export default SideBar;

