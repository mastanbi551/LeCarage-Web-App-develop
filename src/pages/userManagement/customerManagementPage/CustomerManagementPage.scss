@import "../../../styles/variables.scss";

.customer-container {
  padding: 95px 20px;
  background: $secondaryWhite;

  &__header-content {
    display: flex;
    padding-bottom: 25px;

    &__heading {
      display: inline-block;
      float: left;
      font-size: 36px !important;
      @include PageHeading;
      display: flex;
      align-items: center;
      margin-left: 17px;
    }

    &__button {
      font-size: 14px;
      display: inline-block;
      float: right;
      right: 20px;
      padding: 15px;
    }
  }

  &__table-content {
    display: block;
    background: $White;
    border-radius: 3px 3px 0px 0px;
  }

  &__table {
    padding: 0px;
    text-align: left;
    thead {
      box-shadow: inset 0px 1px 0px $secondaryLightGrey;
      th {
        padding: 15px 0px 15px 17px;
      }
    }
    tbody {
      .active-icon {
        @include greenStatusIcon;
      }

      .inactive-icon {
        @include redStatusIcon;
      }
      td {
        padding: 15px 0px 15px 17px;
        cursor: pointer;
      }
    }
  }
}

.overflow {
  overflow: auto;
  height: 400px;
}
