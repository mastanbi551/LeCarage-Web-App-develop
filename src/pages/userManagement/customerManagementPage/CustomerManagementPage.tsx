import React from 'react';
import {
  DynamicDataTable,
  PageHeader,
} from '../../../components';
import './CustomerManagementPage.scss';

export interface CustomerProps {
  'Email Address': string,
  'First Name': string,
  'Last Name': string,
  'Phone Number': string,
  'Status': string,
  'uId': string
}
interface CustomerManagementPageInterfaceProps {
  handleAddUser: () => void;
  columns: { Header: string; accessor: string; }[];
  searchValue: string[];
  onViewHandler: (rowInfo: object) => void;
  onEditHandler: (rowInfo: object) => void;
  onDeleteHandler: (rowInfo: object) => void;
  customers: [];
  errorMessage: string;
  filterValues: {mainMenu: string, subMenus: string[]}[];
  access: string;
  itemsPerPage: number;
  rowsPerPageValueChange: (value: string) => void
}

export const CustomerManagementPage: React.FC<CustomerManagementPageInterfaceProps> = ({
  handleAddUser,
  columns,
  searchValue,
  onViewHandler,
  onEditHandler,
  onDeleteHandler,
  customers,
  errorMessage,
  filterValues,
  access,
  rowsPerPageValueChange,
  itemsPerPage
}) => {

  return (
    <>
      <div className='customer-container'>
        <div className='customer-container__header-content'>
          <PageHeader
            text='Customers'
            name='customer-container__header-content__heading'
            isButtonDisable={true}
            buttonName='Add New'
            buttonClassName='customer-container__header-content__button greenButton btn btn-success'
            buttonOnclick={handleAddUser}
            data-testid='addCustomer'
            access={access}
          />
        </div>

        <div className='customer-container__table-content '>
          <div className='customer-container__table'>
            <DynamicDataTable
              tableHeading=''
              isSearchIconEnable={true}
              isFilterIconEnable={true}
              columns={columns}
              data={customers}
              searchValue={searchValue}
              filterValues={filterValues}
              onViewHandler={onViewHandler}
              onEditHandler={onEditHandler}
              onDeleteHandler={onDeleteHandler}
              defaultItemsPerPage={itemsPerPage}
              errorMessage={errorMessage}
              isEditIconEnable={true}
              isDeleteIconEnable={true}
              access={access}
              rowsPerPageValueChange={rowsPerPageValueChange}
            />
          </div>
        </div>
      </div>
    </>
  );
};
