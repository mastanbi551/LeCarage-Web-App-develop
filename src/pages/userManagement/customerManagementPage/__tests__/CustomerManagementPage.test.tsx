import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { CustomerManagementPage } from '../CustomerManagementPage';
import { DynamicDataTable, PageHeader } from '../../../../components';

describe('CustomerManagementPage', () => {
  const props: any = {
    handleAddUser: jest.fn(),
    columns: [],
    searchValue: '',
    onViewHandler: jest.fn(),
    onEditHandler: jest.fn(),
    onDeleteHandler: jest.fn(),
    customers: [],
  };

  const columnsMock = [    { headerText: 'Name', key: 'name' },    { headerText: 'Description', key: 'description' },  ];

  const handleAddUserMock = jest.fn();
  const onViewHandlerMock = jest.fn();
  const onEditHandlerMock = jest.fn();
  const onDeleteHandlerMock = jest.fn();

  const customerMock = [    { name: 'John Doe', email: 'john.doe@example.com', role: 'admin' },    { name: 'Jane Doe', email: 'jane.doe@example.com', role: 'user' },  ];

  it('renders correctly', () => {
    const { asFragment } = render(<CustomerManagementPage {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with search value correctly', () => {
    const tree = renderer
      .create(<CustomerManagementPage {...props} searchValue="John" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls handleAddUser when add user button is clicked', () => {
    const component = renderer.create(<CustomerManagementPage {...props} />);
    const button = component.root.findAll(
      (el) => el.props.className === 'viewButton'
    )[0];

    if (button) {
      button.props.onClick();
      expect(props.onViewHandler).toHaveBeenCalled();
    } else {
      null;
    }
  });

  it('calls onEditHandler when edit button is clicked', () => {
    const component = renderer.create(<CustomerManagementPage {...props} />);
    const button = component.root.findAll(
      (el) => el.props.className === 'editButton'
    )[0];

    if (button) {
      button.props.onClick();
      expect(props.onEditHandler).toHaveBeenCalled();
    } else {
      null;
    }
  });

  it('calls onDeleteHandler when delete button is clicked', () => {
    const component = renderer.create(<CustomerManagementPage {...props} />);
    const button = component.root.findAll(
      (el) => el.props.className === 'deleteButton'
    )[0];

    if (button) {
      button.props.onClick();
      expect(props.onDeleteHandler).toHaveBeenCalled();
    } else {
      null;
    }
  });
});
