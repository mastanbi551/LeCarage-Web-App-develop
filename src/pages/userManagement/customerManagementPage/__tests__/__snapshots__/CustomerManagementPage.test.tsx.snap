// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerManagementPage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="customer-container"
  >
    <div
      class="customer-container__header-content"
    >
      <h1
        class="customer-container__header-content__heading"
      >
        Customers
      </h1>
      <button
        class="customer-container__header-content__button greenButton btn btn-success"
        type="button"
      >
        Add New
      </button>
    </div>
    <div
      class="customer-container__table-content "
    >
      <div
        class="customer-container__table"
      >
        <div
          class="text-align-center norecord"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CustomerManagementPage renders with search value correctly 1`] = `
<div
  className="customer-container"
>
  <div
    className="customer-container__header-content"
  >
    <h1
      className="customer-container__header-content__heading"
    >
      Customers
    </h1>
    <button
      className="customer-container__header-content__button greenButton btn btn-success"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      Add New
    </button>
  </div>
  <div
    className="customer-container__table-content "
  >
    <div
      className="customer-container__table"
    >
      <div
        className="text-align-center norecord"
      />
    </div>
  </div>
</div>
`;
